packages/interface-internal/src/registrar.ts

```diff
@@ -21,46 +26,75 @@ export type {
    */
   StreamHandlerRecord
 }
+/**
+ * The `Registrar` module provides an interface for managing protocol handlers
+ * and topologies in a libp2p network. It enables registering and managing
+ * protocol-specific handlers, ensuring efficient peer-to-peer communication.
+ */
+/**
+ * The `Registrar` interface allows modules to register, manage, and remove
+ * protocol handlers and topology discovery mechanisms in libp2p.
+ */
 
 export interface Registrar {
   /**
-   * Return the list of protocols with registered handlers
+   * Retrieve the list of registered protocol handlers.
+   *
+   * @returns An array of protocol strings.
    */
   getProtocols(): string[]
 
   /**
-   * Add a protocol handler
+   * Register a handler for a specific protocol.
+   *
+   * @param protocol - The protocol string (e.g., `/my-protocol/1.0.0`).
+   * @param handler - The function that handles incoming streams.
+   * @param options - Optional configuration options for the handler.
+   * @returns A promise that resolves once the handler is registered.
    */
   handle(protocol: string, handler: StreamHandler, options?: StreamHandlerOptions): Promise<void>
 
   /**
-   * Remove a protocol handler
+   * Remove a registered protocol handler.
+   *
+   * @param protocol - The protocol to unhandle.
+   * @returns A promise that resolves once the handler is removed.
    */
   unhandle(protocol: string): Promise<void>
 
   /**
-   * Return the handler for the passed protocol
+   * Retrieve the registered handler for a given protocol.
+   *
+   * @param protocol - The protocol to query.
+   * @returns A `StreamHandlerRecord` containing the handler and options.
    */
   getHandler(protocol: string): StreamHandlerRecord
 
   /**
-   * Register a topology handler for a protocol - the topology will be
-   * invoked when peers are discovered on the network that support the
-   * passed protocol.
+   * Register a topology handler for a specific protocol.
    *
-   * An id will be returned that can later be used to unregister the
```

Comment:
This part about the id being used to unregister the topology later is important, please do not delete it.